inc/zoo/swar/SWAR.h

```diff
@@ -362,6 +365,39 @@ constantIsGreaterEqual_MSB_off(SWAR<NBits, T> subtrahend) noexcept {
     }
 }
 
+template<typename T, typename U, typename V>
+constexpr T median(T x, U y, V z) {
+  return (x | y) & (y | z) & (x | z);
+}
+
+template<int NBits, typename T>
+constexpr BooleanSWAR<NBits, T>
+greaterEqual(SWAR<NBits, T> left, SWAR<NBits, T> right) noexcept {
+    // Adapted from TAOCP V4 P152
+    // h is msbselector, x is right, l is lower/left.  Sets MSB to 1 in lanes
+    // in test variable t for when xi < yi for lane i . Invert for greaterEqual.
+    // t = h & ~<x~yz>
+    // z = (x|h) - (y&~h)
+    using S = swar::SWAR<NBits, T>;
+    const auto h = S::MostSignificantBit, x = left.value(), y = right.value();  // x=left, y= right is x < y
+    const auto z = (x|h) - (y&~h);
+    // bitwise ternary median! 
+    const auto t = h & ~median(x, ~y, z);
+    return ~BooleanSWAR<NBits, T>{static_cast<T>(t)};  // ~(x<y) === x >= y
+}
+
+// In the condition where only MSBs will be on, we can fast lookup with 1 multiply the index of the leftmost byte.
+// This appears to be a mapping from the (say) 256 unique values of a 64 bit int where only MSBs of each 8 bits can be on, but I don't fully understand it.
+// Adapted from TAOCP Vol 4A Page 153 Eq 94.
+template<int NBits, typename T>
+constexpr T
+fastFirstLeftOn(SWAR<NBits, T> test) noexcept {
+    const auto width = sizeof(T) * 8;
+    const auto tval = (1ull<<(width-NBits))-1, bval = (1ull<<(NBits-1))-1;
```

Comment:
Let us not initialize constants to 64 bit `unsigned`s, as in `1ull`, it is a bug lurking when we use anything other than integers as the base type of `SWAR`, we should be very watchful to initialize these constants to `T{1}` for example.
Also, it may be performance disadvantageous if the base type is not 64 bits, since perhaps the use of `uint64_t` here might trigger a promotion to 64 bits and a demotion to whatever `T` is.

Reply:
Also, please give it a try to naming the compile time constants with `BactrianCase` instead of `dromedaryCase`

Reply:
Good catch, I got sloppy on the constant inits.